blog/content/post/2020-02-19-opnsense-xen-trenchboot-demo.md

```diff
@@ -0,0 +1,171 @@
+simply not usable due to lack of specific driver - Compex WLE600VX is a good
+example. Our idea is to run `Xen` with `Dom0` and `OPNSense` in virtual machine. What
+we gain with that approach? Dom0, as privileged domain, has direct access to
+hardware. Also, our custom and adapted build of it, supports WLE600VX WiFi card.
+OPNSense supports popular Ethernet Network Interfaces (e.g. Intel E1000). We
+just fasten it together by passing WLE600VX WiFi from Dom0 to OPNSense as
+emulated Intel E1000 NIC. It is so simple and so effective. For imagination of
+that configuration, look at the diagram below.
+
+![coreboot_trenchboot](../../static/img/wle600-xen-opn.png)
+
+As a result, we have reliable and working WiFi Card in router - although
+OPNSense doesn't directly support it. There is also another benefit achieved
+by virtualization. It introduces another level of security. Closing entities
+(such as network interface, firewall software or operating system) in separate
+environments minimizes the effects of a potential attack. User can be sure that
+virus doesn't spread out to entire system, but stay only in infected part.
+
+The idea of virtualization and separation is well implemented in **Qubes OS**.
+If you have never heard about it, it is an operating system, which mainly focus
+on security. To do that, it is using virtualization and isolation - with Xen and
+Dom0. All running environments are user-depended and can be assigned to a
+specific group. Each group has another level of trust and thus greater or
+smaller exposure to attacks. Qubes OS proves that virtualization and isolation
+technology can be successfully used in big projects. Solutions applied and
+verified there can be surely implemented in smaller projects then - such as home
+router.
+
+## Secure updates
+
+Last assumption we have made is to have possibility to update firmware and
+software from trusted source. Many users are afraid of upgrading their firmware
+in case of possible platform damage. Second factor is safety. Who can ensure
+that newer version comes directly from BIOS vendor and is not already
+'infected'? Luckily, there are platforms and tools which can provide it -
+`fwupdmgr` with `Linux Vendor Firmware Service`. Shortly speaking, using them
+gives user an assurance that an upgrade is safe and can be done almost without
+effort. Of course, vendors can't sign up to trusted list just like that. Each
+must complete verification process and only then can be marked as confirmed and
+trusted vendor.
+
+For our platform, secure update feature is a supplement of other ones. With it,
+user get product which is not only reliable and secure, but also can be easily
+managed and always kept up-to-date.
+
+>If you are interested, what exactly is hidden behind fwupd concept, I refer to
+our [LVFS and fwupd blog
+post](https://blog.3mdeb.com/2019/2019-07-11-how-to-safely-and-easily-update-your-firmware/).
+
+## Our implementation - final demo
+
+
+## Summary
```

Comment:
@piotr-kleins here we need pitch that we can provide such image for any platform if someone is interested in. Also maybe it would be great to mention how dom0 would ba upgraded and build?

Reply:
Also there are no system diagrams and execution flows.